Client/Frontend/Brave Notifications/Brave Rewards/RewardsNotification.swift

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveCore
+import BraveShared
+import Shared
+
+public enum RewardsNotificationAction {
+  /// The user opened the ad by either clicking on it directly or by swiping and clicking the "view" button
+  case opened
+  /// The user swiped the ad away
+  case dismissed
+  /// The user ignored the ad for a given amount of time for it to automatically dismiss
+  case timedOut
+  /// The user clicked the thumbs down button by swiping on the ad
+  case disliked
+}
+
+public class RewardsNotification: NSObject, BraveNotification {
+  public var view: UIView
+  public var dismissAction: (() -> Void)?
+  public var id: String { ad.uuid }
+  public let ad: AdNotification
+  public var isHorizontalSwipe: Bool {
+    guard let adView = view as? AdView else { return false }
+    return adView.swipeTranslation != 0
+  }
+  public var dismissPolicy: DismissPolicy = {
+    var dismissTimeInterval: TimeInterval = 30
+    if !AppConstants.buildChannel.isPublic, let override = Preferences.Rewards.adsDurationOverride.value, override > 0 {
+      dismissTimeInterval = TimeInterval(override)
+    }
+    return .automatic(after: dismissTimeInterval)
+  }()
+  
+  private let rewardsHandler: (RewardsNotificationAction) -> Void
+  
+  public func willDismiss(timedOut: Bool) {
+    guard let adView = view as? AdView else { return }
+    adView.setSwipeTranslation(0, animated: true)
+    rewardsHandler(timedOut ? .timedOut : .dismissed)
+  }
+  
+  init(
+    ad: AdNotification,
+    rewardsHandler: @escaping (RewardsNotificationAction) -> Void
```

Comment:
Doesn't need to be called `rewardsHandler` since this class is specifically for rewards, can just be called `handler` or `actionHandler`